badge_server/main.py

```diff
@@ -76,518 +63,195 @@ def _get_pair_status_for_packages(pkg_sets):
-        is_github = True
-
-    commit_number = badge_utils._calculate_commit_number(package_name)
-
-    force_run_check = flask.request.args.get('force_run_check')
-    # Remove the last '/' from the url root
-    url_prefix = flask.request.url_root[:-1]
-    # Call the url for each badge to run the checks. This will populate the
-    # individual caches, which are used to calculate the final image state.
-    # Self compatibility badge
-    requests.get(url_prefix + flask.url_for(
-        'self_compatibility_badge_image',
-        package=package_name,
-        force_run_check=force_run_check,
-        commit_number=commit_number))
-    # Google compatibility badge
-    requests.get(url_prefix + flask.url_for(
-        'google_compatibility_badge_image',
-        package=package_name,
-        force_run_check=force_run_check,
-        commit_number=commit_number))
-    # Self dependency badge
-    requests.get(url_prefix + flask.url_for(
-        'self_dependency_badge_image',
-        package=package_name,
-        force_run_check=force_run_check,
-        commit_number=commit_number))
-
-    status, timestamp, _, _, _ = _get_all_results_from_cache(
-        package_name, commit_number=commit_number)
-    color = badge_utils.STATUS_COLOR_MAPPING[status]
-
-    details_link = url_prefix + flask.url_for('one_badge_target',
-                                              package=package_name)
-
-    # Include the check timestamp for github head
-    if is_github and timestamp:
-        badge_name = '{} {}'.format(badge_name, timestamp)
-
-    response = flask.make_response(
-        pybadges.badge(
-            left_text=badge_name,
-            right_text=status,
-            right_color=color,
-            whole_link=details_link))
-    response.content_type = badge_utils.SVG_CONTENT_TYPE
-    response.headers['Cache-Control'] = 'no-cache'
-    response.add_etag()
-
-    return response
+    return timestamp
 
 
-@app.route('/one_badge_target')
-def one_badge_target():
-    package_name = flask.request.args.get('package')
-    commit_number = badge_utils._calculate_commit_number(package_name)
-
-    status, _, self_compat_res, google_compat_res, dependency_res = \
-        _get_all_results_from_cache(package_name, commit_number)
-
-    return flask.render_template(
-        'one-badge.html',
-        package_name=package_name,
-        self_compat_res=self_compat_res,
-        google_compat_res=google_compat_res,
-        dependency_res=dependency_res,
-        commit_number=commit_number)
-
-
-@app.route('/self_compatibility_badge_image')
-def self_compatibility_badge_image():
-    """Badge showing whether a package is compatible with itself."""
-    package_name = flask.request.args.get('package')
-    force_run_check = flask.request.args.get('force_run_check')
-    commit_number = flask.request.args.get('commit_number')
+def _get_results_from_compatibility_store(
+        pkgname: str,
+        commit_number: str = None):
```

Comment:
Is the `commit_number` used anywhere?

Reply:
Not in this PR, but it will be needed to determine if the data returned from compatibility store is stale or not for the github repo use case. Let's keep it for now since I'll be adding this functionality soon.

Reply:
The github API has a rate limit: https://developer.github.com/v3/#rate-limiting

We might need to look into other ways to find out the latest commit number, but it's probably fine to keep using this for now.

Reply:
SGTM